From 393ddb4595da77eedc69bcef1319cdd4cb49701b Mon Sep 17 00:00:00 2001
From: Diane Trout <diane@ghic.org>
Date: Fri, 17 Mar 2017 10:37:26 -0700
Subject: [PATCH 18/36] Set default empty structure on Application and Element
 MesssagePtr

Fix for the fix in Bug 746584.

The fix that was committed causes one test to not build.

At some point it was pointed out I shouldn't arbitrarily break source
compatibility, so this fixes the build error and makes code written
before commit 0384481b2f670b3db1a147a1ac0c5ea24c66554e work
---
 src/QGst/message.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/QGst/message.h b/src/QGst/message.h
index 2d66ba1..7ed2bdf 100644
--- a/src/QGst/message.h
+++ b/src/QGst/message.h
@@ -205,7 +205,7 @@ class QTGSTREAMER_EXPORT ApplicationMessage : public Message
     QGST_WRAPPER_FAKE_SUBCLASS(Application, Message)
 public:
     static ApplicationMessagePtr create(const ObjectPtr & source,
-                                        const Structure & structure);
+                                        const Structure & structure=Structure());
 };
 
 /*! \headerfile message.h <QGst/Message>
@@ -216,7 +216,7 @@ class QTGSTREAMER_EXPORT ElementMessage : public Message
     QGST_WRAPPER_FAKE_SUBCLASS(Element, Message)
 public:
     static ElementMessagePtr create(const ObjectPtr & source,
-                                    const Structure & structure);
+                                    const Structure & structure=Structure());
 };
 
 //maybe do: SEGMENT_START (internal)
-- 
2.26.2

